[gradle] Non-existent extra source set directories cause build to fail

<!-- Thanks for filing an issue! Please use this template to provide more context for the issue . -->

**Description of the issue**: When the project registers extra source set directories but the folder doesn't exist, the build fails with

```
Step #3 - "curio-generated-build-all": Caused by: com.google.cloud.tools.jib.plugins.common.BuildStepsExecutionException: Build image failed
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build(BuildStepsRunner.java:269)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.gradle.BuildImageTask.buildImage(BuildImageTask.java:119)
Step #3 - "curio-generated-build-all": at jdk.internal.reflect.GeneratedMethodAccessor804.invoke(Unknown Source)
Step #3 - "curio-generated-build-all": at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Step #3 - "curio-generated-build-all": at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
Step #3 - "curio-generated-build-all": ... 50 more
Step #3 - "curio-generated-build-all": Caused by: java.nio.file.NoSuchFileException: /workspace/frontend/server/src/main/proto
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate.<init>(LayerEntriesSelector.java:72)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector.toSortedJsonTemplates(LayerEntriesSelector.java:129)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector.generateSelector(LayerEntriesSelector.java:145)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.Cache.retrieve(Cache.java:98)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:107)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:37)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
```

**Expected behavior**: The image build succeeds because a non-existent directory is not a fatal error (even if it does indicate a possible project misconfiguration)

**Steps to reproduce**:

```bash
git clone https://github.com/anuraaga/jib-issue-1271
cd jib-issue-1271
./gradlew jibBuildTar
```

**Environment**: Linux, Gradle 5.0-rc-4, Java 11, gradle-jib-plugin 0.10.0

**`jib-gradle-plugin` Configuration**:
```groovy
apply plugin: 'com.google.protobuf'
apply plugin: 'com.google.cloud.tools.jib'
```

**Log output**: <!-- If applicable, provide relevant log output -->

```
Step #3 - "curio-generated-build-all": Caused by: com.google.cloud.tools.jib.plugins.common.BuildStepsExecutionException: Build image failed
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.plugins.common.BuildStepsRunner.build(BuildStepsRunner.java:269)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.gradle.BuildImageTask.buildImage(BuildImageTask.java:119)
Step #3 - "curio-generated-build-all": at jdk.internal.reflect.GeneratedMethodAccessor804.invoke(Unknown Source)
Step #3 - "curio-generated-build-all": at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Step #3 - "curio-generated-build-all": at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:73)
Step #3 - "curio-generated-build-all": ... 50 more
Step #3 - "curio-generated-build-all": Caused by: java.nio.file.NoSuchFileException: /workspace/frontend/server/src/main/proto
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector$LayerEntryTemplate.<init>(LayerEntriesSelector.java:72)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector.toSortedJsonTemplates(LayerEntriesSelector.java:129)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.LayerEntriesSelector.generateSelector(LayerEntriesSelector.java:145)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.cache.Cache.retrieve(Cache.java:98)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:107)
Step #3 - "curio-generated-build-all": at com.google.cloud.tools.jib.builder.steps.BuildAndCacheApplicationLayerStep.call(BuildAndCacheApplicationLayerStep.java:37)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.TrustedListenableFutureTask$TrustedFutureInterruptibleTask.runInterruptibly(TrustedListenableFutureTask.java:125)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.InterruptibleTask.run(InterruptibleTask.java:57)
Step #3 - "curio-generated-build-all": at com.google.common.util.concurrent.TrustedListenableFutureTask.run(TrustedListenableFutureTask.java:78)
```

**Additional Information**: <!-- Any additional information that may be helpful -->


<!-- Thanks for contributing! -->

--- a/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java
+++ b/jib-gradle-plugin/src/main/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurations.java
@@ -131,12 +131,16 @@ class GradleLayerConfigurations {
 
     // Adds dependency files.
     for (File dependencyFile : dependencyFiles) {
-      boolean isSnapshot = dependencyFile.getName().contains("SNAPSHOT");
-      LayerType layerType = isSnapshot ? LayerType.SNAPSHOT_DEPENDENCIES : LayerType.DEPENDENCIES;
-      layerBuilder.addFile(
-          layerType,
-          dependencyFile.toPath(),
-          dependenciesExtractionPath.resolve(dependencyFile.getName()));
+      if (dependencyFile.exists()) {
+        boolean isSnapshot = dependencyFile.getName().contains("SNAPSHOT");
+        LayerType layerType = isSnapshot ? LayerType.SNAPSHOT_DEPENDENCIES : LayerType.DEPENDENCIES;
+        layerBuilder.addFile(
+            layerType,
+            dependencyFile.toPath(),
+            dependenciesExtractionPath.resolve(dependencyFile.getName()));
+      } else {
+        logger.info("\t'" + dependencyFile + "' (not found, skipped)");
+      }
     }
 
     // Adds all the extra files.

--- a/jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurationsTest.java
+++ b/jib-gradle-plugin/src/test/java/com/google/cloud/tools/jib/gradle/GradleLayerConfigurationsTest.java
@@ -212,6 +212,22 @@ public class GradleLayerConfigurationsTest {
   }
 
   @Test
+  public void test_missingDependencyFiles() throws IOException {
+    Path nonexistentFile = Paths.get("/nonexistent/dependency");
+    FileCollection runtimeFileCollection =
+        new TestFileCollection(Collections.singleton(nonexistentFile));
+    Mockito.when(mockMainSourceSet.getRuntimeClasspath()).thenReturn(runtimeFileCollection);
+
+    AbsoluteUnixPath appRoot = AbsoluteUnixPath.get("/app");
+    GradleLayerConfigurations.getForProject(
+        mockProject, mockLogger, extraFilesDirectory, Collections.emptyMap(), appRoot);
+
+    Mockito.verify(mockLogger)
+        .info("Adding corresponding output directories of source sets to image");
+    Mockito.verify(mockLogger).info("\t'" + nonexistentFile + "' (not found, skipped)");
+  }
+
+  @Test
   public void test_extraFiles() throws IOException {
     JavaLayerConfigurations javaLayerConfigurations =
         GradleLayerConfigurations.getForProject(

